crates/hir-def/src/find_path.rs

```diff
@@ -70,45 +67,41 @@ pub enum PrefixKind {
     /// This is the same as plain, just that paths will start with `self` prepended if the path
     /// starts with an identifier that is not a crate.
     BySelf,
-    /// Causes paths to ignore imports in the local module.
+    /// Causes paths to not use a self, super or crate prefix.
     Plain,
     /// Causes paths to start with `crate` where applicable, effectively forcing paths to be absolute.
     ByCrate,
 }
 
 impl PrefixKind {
     #[inline]
-    fn prefix(self) -> PathKind {
+    fn path_kind(self) -> PathKind {
         match self {
             PrefixKind::BySelf => PathKind::Super(0),
             PrefixKind::Plain => PathKind::Plain,
             PrefixKind::ByCrate => PathKind::Crate,
         }
     }
-
-    #[inline]
-    fn is_absolute(&self) -> bool {
-        self == &PrefixKind::ByCrate
-    }
 }
 
 #[derive(Copy, Clone)]
 struct FindPathCtx<'db> {
     db: &'db dyn DefDatabase,
-    prefixed: Option<PrefixKind>,
+    prefix: PrefixKind,
     prefer_no_std: bool,
     prefer_prelude: bool,
+    ignore_local_imports: bool,
```

Comment:
i suspect that https://github.com/rust-lang/rust-analyzer/pull/17252 might help here: i found myself getting confused by the booleans in other contexts 😅

Reply:
Right, guess I'll fix up the conflicts after merging this then 😬